Add JMS bbes (#4880)
ballerina-platform/ballerina-distribution: examples/jms-producer-transaction/jms_producer_transaction.md

# JMS message producer - Transactions

> The `jms:MessageProducer` can become a transactional producer by `'commit'` and `'rollback` functionality of `jms:Session`. Upon successful execution of the transaction block, the `jms:Session` can commit or roll back in the case of any error. Use this to send messages atomically to a JMS provider.

Reviewer: ```suggestion
The `jms:MessageProducer` can become a transactional producer via the `'commit'` and `'rollback` functionalities of the `jms:Session`. Upon successful execution of the transaction block, the `jms:Session` can commit or roll back in the case of any error. Use this to send messages atomically to a JMS provider.
```

::: code jms_producer_transaction.bal :::

> ## Prerequisites

Reviewer: ```suggestion
## Prerequisites
```
Ditto

> - Start a [ActiveMQ broker](https://activemq.apache.org/getting-started) instance.

Reviewer: ```suggestion
- Start a [ActiveMQ broker](https://activemq.apache.org/getting-started) instance.
```
Ditto

Run the program by executing the following command.

::: out jms_producer_transaction.out :::

Invoke the service by executing the following cURL command in a new terminal.

::: out jms_producer_transaction.curl.out :::

## Related links
- [`jms:Session->'commit` function - API documentation](https://lib.ballerina.io/ballerinax/java.jms/latest#Session-commit)

> - [`jms:Session->'rollback` function - API documentation](https://lib.ballerina.io/ballerinax/java.jms/latest#Session-rollback)

Reviewer 2: I can't find API doc for 'commit and 'rollback command?

Author: This will be added in the coming release. Jst added dummy URL

Author: These APIs are newly introduced for SL update 8. Hence, added the expected URL.
